sphere.h

```diff
@@ -7,16 +7,33 @@
 
 
 /**
- * @brief 
+ * A representation of the spheres present in the game board.
+ * 
+ * Extends the map class, as we want to use most of their functionality.
+ * However, tihs class allows us to have constructors that ensure that the
+ * map has keys for all the different elements and no extraneous keys,
+ * as well as constructors that allow various formats of the number of gems present
+ * in a sphere to control how many of each type of gem are in the sphere when
+ * created. 
```

Comment:
I think we can word this a little better. Something like:
```suggestion
 * This class allows us to have constructors that ensure that the
 * map has keys for all the different elements and no extraneous keys,
 * as well as constructors that allow various formats of the number of gems present
 * in a sphere to control how many of each type of gem are in the sphere when
 * created. 
```